x-pack/plugins/alerting/common/maintenance_window.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger, SavedObjectsClientContract } from '@kbn/core/server';
+import { RRuleParams } from './rrule_type';
+
+export enum MaintenanceWindowStatus {
+  Running = 'running',
+  Upcoming = 'upcoming',
+  Finished = 'finished',
+  Archived = 'archived',
+}
+
+export interface MaintenanceWindowModificationMetadata {
+  createdBy: string | null;
+  updatedBy: string | null;
+  createdAt: string;
+  updatedAt: string;
+}
+
+export interface DateRange {
+  gte: string;
+  lte: string;
+}
+
+export interface MaintenanceWindowSOProperties {
+  title: string;
+  enabled: boolean;
+  duration: number;
+  expirationDate: string;
+  events: DateRange[];
+  rRule: RRuleParams;
+}
+
+export type MaintenanceWindowSOAttributes = MaintenanceWindowSOProperties &
+  MaintenanceWindowModificationMetadata;
+
+export type MaintenanceWindow = MaintenanceWindowSOAttributes & {
+  status: MaintenanceWindowStatus;
+  startDate: string | null;
+  endDate: string | null;
```

Comment:
We have these start and end dates that map to: 
![image](https://user-images.githubusercontent.com/74562234/228920165-befd8b77-141f-4c52-a1c7-7a708af4717c.png)

The dates indicate the start and end date of the current or upcoming EVENT, not the start and end date of the entire maintenance window schedule, should I name these something like `eventStartDate` instead?

Reply:
`eventStartDate` sounds good to me, I agree that start and end date could be confusing